### PR TITLE
#32996 support for new 2016 ext proxy settings

### PR DESCRIPTION
**Pull request by manneohrstrom**
Implements new 2016 extension 1 proxy settings in the flame project creator.

When running older versions of Flame, the old settings UI is displayed.
When 2016 ext 1 and above is being used, a Proxy UI similar to that in Flame is being used:

image

Also resolves some other minor issues:

- If the default log location `/usr/discreet/log` cannot be written to, logging falls back to using `/tmp/`
- If wiretap cannot connected to, an improved error message is displayed. Previously, it would say "cannot get number of children".

### CODE REVIEW

File: engine.py [hunk 1]
```diff
@@ -128,6 +151,10 @@ def _initialize_logging(self):
             logger.setLevel(logging.DEBUG)
         else:
             logger.setLevel(logging.INFO)
+
+        if using_safe_log_file:
+            logger.error("Cannot write to standard log file location %s! As a fallback, "
+                         "logs will be written to %s instead." % (self.SGTK_LOG_FILE, log_file))
```
Comment (josh-t) [inline engine.py]: is it worth saying that this is typically due to permission issues?

File: python/tk_flame/project_create_dialog.py [hunk 2]
```diff
@@ -120,11 +150,59 @@ def __init__(self,
...
+        Populate the proxy UI tab with values from 
+        the given project settings dictionary. This method is for
+        flame versions 2016.1 and above
+        
+        :param project_settings: Dictionary as returned by 
+                                 the project_startup_hook's get_project_settings method
+        """
+        
+        # first hide the old proxy tab
+        self.ui.tabWidget.removeTab(self.TAB_OLD_PROXY)
```
Comment (josh-t) [inline python/tk_flame/project_create_dialog.py]: minor/picky, but would it be clearer to put the tab removal calls into the conditional that calls these set up methods? just thinking that the removal of those is the opposite of "set up" and its also dealing with a different tab. :)

File: python/tk_flame/project_create_dialog.py [hunk 3]
```diff
@@ -120,11 +150,59 @@ def __init__(self,
...
+            self.ui.new_generate_proxies.setChecked(True)
+        else:
+            self.ui.new_generate_proxies.setChecked(False)
+        
+        # set the minimum size slider
+        proxy_min_frame_size = int(project_settings.get("ProxyMinFrameSize") or 0)
+        self.ui.new_proxy_min_frame_size.setValue(proxy_min_frame_size)
+        self.ui.new_proxy_min_frame_size.valueChanged.connect(self._on_new_proxy_min_frame_size_change)
+        
+        # refresh value at startup
```
Comment (josh-t) [inline python/tk_flame/project_create_dialog.py]: maybe a quick comment about why this is necessary?

Comment (josh-t) [inline python/tk_flame/project_create_dialog.py]: oh i see, it looks like it's updating a label. maybe just say something like "make sure the preview label is updated to display the value at startup"?

File: python/tk_flame/wiretap.py [hunk 4]
```diff
@@ -266,6 +274,43 @@ def _ensure_project_exists(self, project_name, user_name, workspace_name):
                
             self._engine.log_debug( "Project successfully created.")
          
+         
+         
+    def _append_setting_to_xml(self, project_settings, setting, starts_working_in=None, stops_working_in=None):
+        """
+        Generates xml for a parameter. May return an empty string if the xml
+        cannot or should not be generated.
+
```
Comment (josh-t) [inline python/tk_flame/wiretap.py]: cool method! i like it. :)

**Comment by josh-t** [comment]
Only a few small comments. Overall looks great! Good to merge whenever.

File: engine.py (hunk 1 above)
Comment (manneohrstrom) [inline engine.py]: good point! adding!

File: python/tk_flame/project_create_dialog.py (hunk 2 above)
Comment (manneohrstrom) [inline python/tk_flame/project_create_dialog.py]: good point, i ended up moving it into the main method, seemed clearer that way

File: python/tk_flame/project_create_dialog.py (hunk 3 above)
Comment (manneohrstrom) [inline python/tk_flame/project_create_dialog.py]: much better comment! Updated!

**Pull request closed without merging**